dissect/target/plugins/os/windows/notifications.py

```diff
@@ -55,19 +192,172 @@ def __init__(self, target):
         self.appdb_files = []
 
         for user_details in target.user_details.all_with_home():
-            notification_dir = user_details.home_path.joinpath("AppData/Local/Microsoft/Windows/Notifications")
+            notification_dir = user_details.home_path.joinpath(NOTIFICATIONS_DIR)
             wpndb_file = notification_dir.joinpath("wpndatabase.db")
             appdb_file = notification_dir.joinpath("appdb.dat")
 
             if wpndb_file.exists():
                 self.wpndb_files.append((user_details.user, wpndb_file))
 
             if appdb_file.exists():
-                self.appdb_files.append((user_details.user, appdb_file))
+                with appdb_file.open(mode="rb") as fp:
+                    chunk = c_appdb.Chunk(fp)
+                    if chunk.Header.Magic == APPDB_MAGIC:
+                        version = chunk.Header.Version
+                        if version == 3:
+                            self.appdb_files.append((user_details.user, appdb_file))
+                        else:
+                            self.target.log.warning(
+                                "Skipping %s: unsupported version %s.",
+                                appdb_file,
+                                version,
+                            )
+                            if version != 1:
+                                self.target.log.warning(
+                                    "Unknown appdb version %s in file %s, "
+                                    "please consider providing us with a sample.",
+                                    version,
+                                    appdb_file,
+                                )
 
     def check_compatible(self) -> None:
         if not self.wpndb_files and not self.appdb_files:
-            raise UnsupportedPluginError("No wpndatabase.db or appdb.dat files found")
+            raise UnsupportedPluginError("No or incompatible wpndatabase.db or appdb.dat files found")
+
+    def _get_appdb_chunk_record(
+        self,
+        chunk: c_appdb.Chunk,
+        timestamp: Optional[datetime.datetime],
+        user: RecordDescriptor,
```

Comment:
```suggestion
        user: WindowsUserRecord,
```
Perhaps we should fix the type hints in the `user_details` plugin.